File: src/Gang.Authentication/Api/GangAuthentication.cs
namespace Gang.Authentication.Api
{
    public sealed class GangAuthentication
    {
        public GangAuthentication(
            string credentialId,
            string clientData,
            string authenticatorData,
            string signature
            )
        {
            CredentialId = credentialId;
            ClientData = clientData;
            AuthenticatorData = authenticatorData;
            Signature = signature;
        }

        public string CredentialId { get; }
        public string ClientData { get; }
        public string AuthenticatorData { get; }
        public string Signature { get; }
    }
}

File: src/GangJs/Core/src/models/authentication/GangAuthentication.ts
import { bytesToBase64Url } from '../../services/utils';

export class GangAuthentication {
  constructor(
    public readonly credentialId: string,
    public readonly clientData: string,
    public readonly authenticatorData: string,
    public readonly signature: string
  ) {}

  static from(
    credentialId: string,
    clientDataJSON: ArrayBuffer,
    authenticatorData: ArrayBuffer,
    signature: ArrayBuffer
  ): GangAuthentication {
    return new GangAuthentication(
      credentialId,
      bytesToBase64Url(clientDataJSON),
      bytesToBase64Url(authenticatorData),
      bytesToBase64Url(signature)
    );
  }
}
